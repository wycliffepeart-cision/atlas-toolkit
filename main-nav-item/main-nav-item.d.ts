export declare class MainNavItem extends HTMLElement {
}
//# sourceMappingURL=main-nav-item.d.ts.map