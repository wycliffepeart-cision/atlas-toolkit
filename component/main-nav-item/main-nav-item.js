import { _ as __decorate } from '../../tslib.es6-2e0713a3.js';
import { Template } from '../../decorators/template.js';

var MainNavItemTemplate = "<template>\n  <slot></slot>\n</template>\n";

let MainNavItem = class MainNavItem extends HTMLElement {
};
MainNavItem = __decorate([
    Template(MainNavItemTemplate)
], MainNavItem);
if (!customElements.get('atlas-main-nav-item'))
    customElements.define('atlas-main-nav-item', MainNavItem);

export { MainNavItem };
