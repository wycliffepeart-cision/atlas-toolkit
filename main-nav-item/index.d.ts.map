{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["../../src/main-nav-item/index.ts"],"names":[],"mappings":""}