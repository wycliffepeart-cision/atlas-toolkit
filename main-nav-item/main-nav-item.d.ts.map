{"version":3,"file":"main-nav-item.d.ts","sourceRoot":"","sources":["../../src/main-nav-item/main-nav-item.ts"],"names":[],"mappings":"AAGA,qBACa,WAAY,SAAQ,WAAW;CAAG"}