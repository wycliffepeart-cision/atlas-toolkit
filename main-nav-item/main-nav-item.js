import { _ as __decorate } from '../tslib.es6-806de115.js';
import { Template } from '../decorators/template.js';

var MainNavItemTemplate = "<template>\n  <slot></slot>\n</template>\n";

let MainNavItem = class MainNavItem extends HTMLElement {
};
MainNavItem = __decorate([
    Template(MainNavItemTemplate)
], MainNavItem);

export { MainNavItem };
