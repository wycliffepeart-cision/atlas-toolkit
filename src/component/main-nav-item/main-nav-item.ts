import { Template } from '../../decorators/template';
import MainNavItemTemplate from './main-nav-item.html';

@Template(MainNavItemTemplate)
export class MainNavItem extends HTMLElement {}

if (!customElements.get('atlas-main-nav-item')) customElements.define('atlas-main-nav-item', MainNavItem);