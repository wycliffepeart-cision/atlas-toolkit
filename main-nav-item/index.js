import { MainNavItem } from './main-nav-item.js';
import '../tslib.es6-806de115.js';
import '../decorators/template.js';

if (!customElements.get('atlas-main-nav-item'))
    customElements.define('atlas-main-nav-item', MainNavItem);
