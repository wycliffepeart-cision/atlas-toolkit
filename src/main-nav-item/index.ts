import { MainNavItem } from './main-nav-item';

if (!customElements.get('atlas-main-nav-item')) customElements.define('atlas-main-nav-item', MainNavItem);
