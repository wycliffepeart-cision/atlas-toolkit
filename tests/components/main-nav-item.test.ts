import '../../src/component/main-nav-item';

const NAV_ITEM_TAG = 'atlas-main-nav-item';
const shadowContentSnapshot = `
	<template>
		<slot></slot>
	</template>
`;

describe('MainNavItem', () => {
  it('is correctly appended in the DOM', async () => {
    await customElements.whenDefined(NAV_ITEM_TAG);
    const MainNavItemElement = document.createElement(NAV_ITEM_TAG);
    document.body.appendChild(MainNavItemElement);

    (expect(MainNavItemElement) as any).toBeInTheDocument();
  });

  it('is capable of being selected', async () => {
    await customElements.whenDefined(NAV_ITEM_TAG);
    const MainNavItemElement = document.createElement(NAV_ITEM_TAG);
    document.body.appendChild(MainNavItemElement);

    const selectedElement = document.querySelector(NAV_ITEM_TAG);
    (expect(selectedElement) as any).toBeInTheDocument();
  });

  it('has the correct content when created', async () => {
    await customElements.whenDefined(NAV_ITEM_TAG);
    const MainNavItemElement: any = document.createElement(NAV_ITEM_TAG);
    document.body.appendChild(MainNavItemElement);

    (expect(MainNavItemElement) as any).toContainHTML('<atlas-main-nav-item></atlas-main-nav-item>');
    (expect(MainNavItemElement) as any).toBeEmptyDOMElement();
    (expect(MainNavItemElement.getShadowDomContent) as any).toMatchSnapshot(shadowContentSnapshot);
  });

  it('contains only one direct child', async () => {
    await customElements.whenDefined(NAV_ITEM_TAG);
    const MainNavItemElement = document.createElement(NAV_ITEM_TAG);
    const directChild = document.createElement('div');
    MainNavItemElement.appendChild(directChild);

    (expect(MainNavItemElement) as any).toContainHTML('<atlas-main-nav-item><div></div></atlas-main-nav-item>');
    (expect(MainNavItemElement.childElementCount) as any).toBe(1);
  });
});
