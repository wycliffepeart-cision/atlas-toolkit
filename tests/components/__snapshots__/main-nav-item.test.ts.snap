// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainNavItem has the correct content when created: 
	<template>
		<slot></slot>
	</template>
 1`] = `undefined`;
